src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let clonedObject = { ...state };
+  const stateWithClones = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        clonedObject = {
+          ...clonedObject,
+          ...action.extraData,
+        };
+        break;
+      case 'removeProperties':
+        clonedObject = { ...clonedObject };
+
+        for (const key of action.keysToRemove) {
+          delete clonedObject[key];
+        }
+        break;
+      case 'clear':
+        clonedObject = {};
+        break;
+      default:
+        return 'Error';
+    }
+    stateWithClones.push(clonedObject);
```

Comment:
`clonedObject` should be spreaded
```suggestion
    stateWithClones.push({ ...clonedObject });
```